### ISSUE TITLE
Customizable Rollups on Payments: sObject Row was retrieved via SOQL without querying the requested field

### ISSUE THREAD

**Issue by jesslopez-sf** [SalesforceFoundation/Cumulus#4330]
Related to #4177 , but not confined to User-Defined Rollups that were converted.

When we create a Payment rollup and the operation does not require an Amount field, we stamp the Payment Amount in the Amount field by default. We do not include the field within our SOQL query but it is then referenced later in the code and throws the following error:

Stack Trace
```
Class.CRLP_Operation_SVC.getAmountValue: line 164, column 1
Class.CRLP_Operation_SVC.doRollupOperation: line 99, column 1
Class.CRLP_RollupContact_SVC.processRollups: line 92, column 1
Class.CRLP_RollupProcessor.doFinalRollup: line 480, column 1
Class.CRLP_RollupProcessor.completeRollupForSingleSummaryRecord: line 450, column 1
Class.CRLP_RollupProcessor.startRollupProcessing: line 352, column 1
Class.CRLP_RollupQueueable.runRollupsForIds: line 170, column 1
Class.CRLP_RollupQueueable.execute: line 73, column 1

```

### Steps to Repeat

_Setup_

1. Create a new Contact Field
    1. Type = Text
    2. Field Label = Last Payment Method
    3. Length = 255
2. Enable Customizable Rollups in NPSP Settings → Donations → Customizable Rollups
3. Create a new Rollup
    1. Target Object = Contact
    2. Target Field = Last Payment Method
    3. Operation = Last
    4. Time Frame = All
    5. Rollup Type = Payment → Contact (Hard Credit)
    6. Filter Group = Pmts: Paid
    7. Field to Roll Up = Payment Method
    8. Date Field = Payment Date

_Test_

1. Create a Contact (First Name = Jess; Last Name = Lopez)
2. Create an Opportunity (Record Type = Donation; Name = Test 1; Account Name= Lopez Household; Amount = $100; Close Date = today, Stage = Closed Won)
3. The following error is captured in the NPSP logs and emailed:
SObject row was retrieved via SOQL without querying the requested field: npe01__OppPayment__c.npe01__Payment_Amount__c

**Comment by jesslopez-sf** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E0000087vgKQAQ" target="_blank">W-033802</a>

**Comment by mrbelvedere** [comment]
Included in beta release 3.158 (Beta 5)

**Comment by mrbelvedere** [comment]
Included in production release 3.158
